src/OVAL/oval_component.c

```diff
@@ -1973,6 +2051,159 @@ static oval_syschar_collection_flag_t _oval_component_evaluate_ESCAPE_REGEX(oval
 	return flag;
 }
 
+#define INITIAL_CAPACITY 64
+
+typedef enum {
+    START,
+    LEFT_BRACKET,
+    CLASS,
+    ESCAPE
+} states;
+
+typedef struct string {
+    char *str;
+    unsigned int length;
+    unsigned int capacity;
+} t_string;
+
+static t_string *_string_new() {
+    t_string *s;
+    s = malloc(sizeof(t_string));
+    if (s == NULL)
+        return NULL;
+    s->str = malloc(INITIAL_CAPACITY);
+    if (s->str == NULL) {
+        free(s);
+        return NULL;
+    }
+    s->str[0] = '\0';
+    s->length = 0;
+    s->capacity = INITIAL_CAPACITY;
+    return s;
+}
+
+static void _string_append(t_string *s, char c)
+{
+    if (s->length + 1 >= s->capacity) {
+        if(realloc(s->str, s->capacity + INITIAL_CAPACITY) == NULL)
+            return;
+        s->capacity += INITIAL_CAPACITY;
+    }
+    s->str[s->length++] = c;
+    s->str[s->length] = '\0';
+}
+
+static char *_glob_to_regex (const char *glob, int noescape)
+{
+    t_string *regex;
+    char * result;
+    char c;
+    int i = 0;
+    int state = START;
+    regex = _string_new();
+    if (regex == NULL) {
+        return NULL;
+    }
+    _string_append(regex, '^');
+    while(1) {
+        c = glob[i++];
+        switch (state) {
+        case START:
+            if (c == '\0') {
+                goto finish;
+            } else if (c == '?') {
+                _string_append(regex, '.');
+            } else if (c == '*') {
+                _string_append(regex, '.');
+                _string_append(regex, '*');
+            } else if (c == '.' || c == '|' || c == '^' || c == '(' || c == ')'
+                    || c == '{' || c == '}' || c == '+' ) {
```

Comment:
What about $, shouldn't that also be escaped? I haven't actually checked the spec, so maybe I am wrong.